internal/connector/proxy.go

```diff
@@ -6,34 +6,49 @@ import (
 	"fmt"
 	"net/http"
 	"net/http/httputil"
+	"strconv"
 	"sync"
+	"time"
 
-	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/infrahq/infra/internal/certs"
 	"github.com/infrahq/infra/internal/logging"
+	"github.com/infrahq/infra/metrics"
 )
 
 func proxyMiddleware(
 	proxy *httputil.ReverseProxy,
 	authn *authenticator,
 	bearerToken string,
-) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		claim, err := authn.Authenticate(c.Request)
+) func(resp http.ResponseWriter, req *http.Request) {
+	return func(resp http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		status := http.StatusOK
+		defer func() {
+			metrics.RequestDuration.With(prometheus.Labels{
```

Comment:
Ah, I think I missed registering this metric. That was previously done with the middleware. Fixing this now.

Reply:
Ok, fixed this by registering the metric in the same place we register the client output request time.